libs/adapters/src/utils/setupErrorHandlers.ts

```diff
@@ -48,6 +55,15 @@ export const setupErrorHandlers = (app: Express) => {
         status: error.status,
         error: error.message,
       });
+    } else if (error instanceof InvalidActor) {
+      log.warn(error.message, { error, reqContext }); // just warn, to avoid overloading rollbar with bots and attacks
+      res.status(401).send({ status: 401, error: error.message });
+    } else if (error instanceof InvalidInput) {
+      log.warn(error.message, { error, reqContext }); // just warn, to avoid overloading rollbar with bots and attacks
+      res.status(400).send({ status: 400, error: error.message });
+    } else if (error instanceof InvalidState) {
+      log.warn(error.message, { error, reqContext }); // just warn, to avoid overloading rollbar with bots and attacks
```

Comment:
500 errors should be logged with `log.error` so they are reported to Rollbar.

Reply:
Right, but invalid state can be a tricky one... basically an attacker can induce invalid states by passing valid inputs... so maybe this should be a 400... let's discuss tomorrow